extensions/blocks/embed/editor.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * External dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+
+/**
+ * Re-activates the Facebook and Instagram embed block variations by
+ * making them appear in the inserter and their URL patterns parseable again.
+ *
+ * Relevant for Gutenberg >= 9.0, where the aforementioned blocks were deprecated*[0] because
+ * they do not support the oEmbed changes that Facebook is going to implement on Oct. 24th, 2020,
+ * and core has currently no plans to support it.
+ *
+ * However, we do plan on keeping support for them in Jetpack and WordPress.com by sending an
+ * access token alongside the oEmbed API request.
+ *
+ * Our goal is for this go unnoticed by our end-users, as this is only an implementation detail.
+ *
+ * *[0] https://github.com/WordPress/gutenberg/pull/24472.
+ *
+ * @param {object} settings - Block settings object.
+ * @param {string} name - The block name
+ * @returns {object} The settings for the given block with the patched variations.
+ */
+function reactivateFacebookAndInstagramEmbedBlockVariations( settings, name ) {
+	if ( name !== 'core/embed' || ! settings.variations ) {
+		return settings;
+	}
+
+	/**
+	 * Given the variation name and the pattern regex (the attribute that was removed from the deprecated
+	 * variations) returns a new "patched" variation object that essentially reverts the deprecation changes
+	 * made in https://github.com/WordPress/gutenberg/pull/24472.
+	 */
+	settings.variations.forEach( function ( variation ) {
+		switch ( variation.name ) {
```

Comment:
I know this is a draft PR, so feel free to ignore this if you're already considering it. When I look at core, I see that there is a bit more information with each embed block.

<img width="962" alt="Screen Shot 2020-09-17 at 12 23 44 PM" src="https://user-images.githubusercontent.com/1126811/93505701-d057d000-f8e0-11ea-9ebf-e2b228d750b8.png">

Will we also need to add this information to this PR?

Reply:
That information should still be present. We're only modifying the FB and IG block variations (that [still exist](https://github.com/WordPress/gutenberg/blob/b34113c6d5eda5c459070c68179b2ed458567e60/packages/block-library/src/embed/variations.js#L58-L85) in GB) to change back what was modifed by https://github.com/WordPress/gutenberg/pull/24472 (which is what removed them from the block inserter, and made them not appear when pasting a matching FB or IG URL) :slightly_smiling_face:

Reply:
Got ya. Thanks for the clarification!